error: 'OpenImageIO::v1_2::spin_mutex' has no member named 'try_lock'

When building OpenImageIO 1.2.3 on i486 with threads disabled I'm getting

```
cd /home/users/builder2/rpm/BUILD/OpenImageIO-oiio-bcdad81/build/libOpenImageIO && /usr/bin/i486-pld-linux-g++   -DBOOST_TEST_DYN_LINK -DNOTHREADS=1 -DOpenImageIO_EXPORTS -DPTEX_EXPORTS -DUSE_FIELD3D=1 -DUSE_FREETYPE -DUSE_OCIO=1 -DUSE_OPENCV -DUSE_OPENEXR_VERSION2=1 -DUSE_OPENSSL=1 -DUSE_TBB=0 -O2 -fwrapv -pipe -Wformat -Werror=format-security -gdwarf-4 -fno-debug-types-section -fvar-tracking-assignments -g2 -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fomit-frame-pointer -march=i486 -fasynchronous-unwind-tables -gdwarf-4 -fno-debug-types-section -fvar-tracking-assignments -g2  -DNDEBUG -DQT_NO_DEBUG -fPIC -I/usr/include/OpenEXR -I/home/users/builder2/rpm/BUILD/OpenImageIO-oiio-bcdad81/src/include -I/home/users/builder2/rpm/BUILD/OpenImageIO-oiio-bcdad81/build/include -I/usr/include/freetype2    -Wall -Werror -o CMakeFiles/OpenImageIO.dir/__/libtexture/imagecache.cpp.o -c /home/users/builder2/rpm/BUILD/OpenImageIO-oiio-bcdad81/src/libtexture/imagecache.cpp
/home/users/builder2/rpm/BUILD/OpenImageIO-oiio-bcdad81/src/libtexture/imagecache.cpp: In member function 'void OpenImageIO::v1_2::pvt::ImageCacheImpl::check_max_files(OpenImageIO::v1_2::pvt::ImageCachePerThreadInfo*)':
/home/users/builder2/rpm/BUILD/OpenImageIO-oiio-bcdad81/src/libtexture/imagecache.cpp:1056:30: error: 'OpenImageIO::v1_2::spin_mutex' has no member named 'try_lock'
     if (! m_file_sweep_mutex.try_lock())
                              ^
/home/users/builder2/rpm/BUILD/OpenImageIO-oiio-bcdad81/src/libtexture/imagecache.cpp: In member function 'void OpenImageIO::v1_2::pvt::ImageCacheImpl::check_max_mem(OpenImageIO::v1_2::pvt::ImageCachePerThreadInfo*)':
/home/users/builder2/rpm/BUILD/OpenImageIO-oiio-bcdad81/src/libtexture/imagecache.cpp:1924:30: error: 'OpenImageIO::v1_2::spin_mutex' has no member named 'try_lock'
     if (! m_tile_sweep_mutex.try_lock())
                              ^
make[2]: *** [libOpenImageIO/CMakeFiles/OpenImageIO.dir/__/libtexture/imagecache.cpp.o] Error 1
```

similar error happens on ppc
https://bugzilla.redhat.com/show_bug.cgi?id=1021977

--- a/src/include/thread.h
+++ b/src/include/thread.h
@@ -128,6 +128,7 @@ public:
     void unlock () { }
     void lock_shared () { }
     void unlock_shared () { }
+    bool try_lock () { return true; }
 };
 
 /// Null lock that can be substituted for a real one to test how much
@@ -219,7 +220,9 @@ using boost::thread_specific_ptr;
 inline int
 atomic_exchange_and_add (volatile int *at, int x)
 {
-#ifdef USE_GCC_ATOMICS
+#ifdef NOTHREADS
+    int r = *at;  *at += x;  return r;
+#elif defined(USE_GCC_ATOMICS)
     return __sync_fetch_and_add ((int *)at, x);
 #elif USE_TBB
     atomic<int> *a = (atomic<int> *)at;
@@ -237,7 +240,9 @@ atomic_exchange_and_add (volatile int *at, int x)
 inline long long
 atomic_exchange_and_add (volatile long long *at, long long x)
 {
-#ifdef USE_GCC_ATOMICS
+#ifdef NOTHREADS
+    long long r = *at;  *at += x;  return r;
+#elif defined(USE_GCC_ATOMICS)
     return __sync_fetch_and_add (at, x);
 #elif USE_TBB
     atomic<long long> *a = (atomic<long long> *)at;
@@ -261,11 +266,17 @@ atomic_exchange_and_add (volatile long long *at, long long x)
 ///        *at = newval;  return true;
 ///    } else {
 ///        return false;
-///
+///    }
 inline bool
 atomic_compare_and_exchange (volatile int *at, int compareval, int newval)
 {
-#ifdef USE_GCC_ATOMICS
+#ifdef NOTHREADS
+    if (*at == compareval) {
+        *at = newval;  return true;
+    } else {
+        return false;
+    }
+#elif defined(USE_GCC_ATOMICS)
     return __sync_bool_compare_and_swap (at, compareval, newval);
 #elif USE_TBB
     atomic<int> *a = (atomic<int> *)at;
@@ -282,7 +293,13 @@ atomic_compare_and_exchange (volatile int *at, int compareval, int newval)
 inline bool
 atomic_compare_and_exchange (volatile long long *at, long long compareval, long long newval)
 {
-#ifdef USE_GCC_ATOMICS
+#ifdef NOTHREADS
+    if (*at == compareval) {
+        *at = newval;  return true;
+    } else {
+        return false;
+    }
+#elif defined(USE_GCC_ATOMICS)
     return __sync_bool_compare_and_swap (at, compareval, newval);
 #elif USE_TBB
     atomic<long long> *a = (atomic<long long> *)at;

